xml-occurrence-parsers/src/main/java/org/gbif/xml/occurrence/parser/parsing/xml/XmlFragmentParser.java

```diff
@@ -208,4 +213,34 @@ public static RawOccurrenceRecord parseRecord(byte[] xml, OccurrenceSchemaType s
 
     return results;
   }
+
+  /**
+   * Filters the records by discarding the ones without ID.
+   */
+  private static List<RawOccurrenceRecord> filterRecords(List<RawOccurrenceRecord> records) {
+    if (records == null) {
+      return Collections.emptyList();
+    }
+
+    List<RawOccurrenceRecord> result = new ArrayList<>(records.size());
+    for (RawOccurrenceRecord record : records) {
+      // try to set an id in case it is missing
+      if (Strings.isNullOrEmpty(record.getId())) {
+        // try to create a triplet
+        Triplet triplet = null;
```

Comment:
Triplet triplet = new Triplet(record.getInstitutionCode(), record.getCollectionCode(), record.getCatalogueNumber());
          record.setId(OccurrenceKeyHelper.toKey(triplet));


or

          record.setId(OccurrenceKeyHelper.toKey(new Triplet(record.getInstitutionCode(), record.getCollectionCode(), record.getCatalogueNumber())));

and remove L239?